cf-java-logging-support-log4j2/src/test/java/com/sap/hcp/cf/log4j2/layout/suppliers/EventContextFieldSupplierTest.java

```diff
@@ -0,0 +1,76 @@
+package com.sap.hcp.cf.log4j2.layout.suppliers;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.logging.log4j.core.LogEvent;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.sap.hcp.cf.log4j2.converter.api.Log4jContextFieldSupplier;
+import com.sap.hcp.cf.log4j2.layout.supppliers.EventContextFieldSupplier;
+import com.sap.hcp.cf.logging.common.customfields.CustomField;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EventContextFieldSupplierTest {
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private LogEvent event;
+
+    private Log4jContextFieldSupplier fieldSupplier = new EventContextFieldSupplier();
+
+    @Test
+    public void emptyMdcAndNoArguments() {
+        when(event.getContextData().toMap()).thenReturn(Collections.emptyMap());
+        Map<String, Object> fields = fieldSupplier.map(event);
+        assertThat(fields.entrySet(), is(empty()));
+    }
+
+    @Test
+    public void mdcFields() throws Exception {
+        HashMap<String, String> mdc = new HashMap<>();
+        mdc.put("key", "value");
+        mdc.put("this", "that");
+        when(event.getContextData().toMap()).thenReturn(mdc);
+
+        Map<String, Object> fields = fieldSupplier.map(event);
+        assertThat(fields, hasEntry("key", "value"));
+        assertThat(fields, hasEntry("this", "that"));
+    }
+
+    @Test
+    public void customFields() throws Exception {
+        Object[] arguments = new Object[] { //
+                                            new Object(), //
+                                            CustomField.customField("key", "value"), //
+                                            CustomField.customField("this", "that") };
+        when(event.getMessage().getParameters()).thenReturn(arguments);
+
+        Map<String, Object> fields = fieldSupplier.map(event);
+        assertThat(fields, hasEntry("key", "value"));
+        assertThat(fields, hasEntry("this", "that"));
```

Comment:
Do we only want to support string values in all contexts?
In application logging the enforcement is clear,
just asking from a cloud logging perspective, as this might be seen as a short coming of this library.

Reply:
I will adjust this to match the logback test case, that contains a numerical value for the second field.